x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/util/DateUtils.java

```diff
@@ -148,4 +150,95 @@ public static int getNanoPrecision(Expression precisionExpression, int nano) {
         nano = nano - nano % (int) Math.pow(10, (9 - precision));
         return nano;
     }
+
+    public static ZonedDateTime truncate(ZonedDateTime dateTime, DateTrunc.DatePart datePart) {
```

Comment:
If the enum is not reused anywhere else (and it doesn't seem to be), I would move the truncate logic inside the enum itself as it looks self-contained.